server/instances.go

```diff
@@ -151,6 +151,29 @@ func (p *Plugin) InstallInstance(instance Instance) error {
 				}
 			}
 
+			if instance.Common().Type == CloudOAuthInstanceType && len(instances.IDs()) > 0 && instances.checkIfExists(instance.GetID()) {
+				p.API.LogDebug("Getting the stored JWT instance data to store it inside the OAuth instance data.")
+				jwtInstance, err := p.instanceStore.LoadInstance(instance.GetID())
+				if err != nil {
+					p.API.LogError("Error occurred while fetching the instance", "ID", instance.GetID(), "Error", err.Error())
+					return err
+				}
+
+				// Storing the JWT instance data inside the OAuth instance data. We will use this to use the stored JWT token in case the user has not connected to OAuth yet.
+				p.API.LogDebug("Instance type stored in KV store", "ID", jwtInstance.GetID(), "Type", jwtInstance.Common().Type)
+				if jwtInstance.Common().Type == CloudInstanceType {
+					instance.(*cloudOAuthInstance).JWTInstance = jwtInstance.(*cloudInstance)
+				} else if jwtInstance.Common().Type == CloudOAuthInstanceType {
+					instance.(*cloudOAuthInstance).JWTInstance = jwtInstance.(*cloudOAuthInstance).JWTInstance
+				}
+
+				if instance.(*cloudOAuthInstance).JWTInstance != nil {
+					p.API.LogDebug("JWT instance is successfully stored inside cloud OAuth instance data.", "JWTInstance", instance.(*cloudOAuthInstance).JWTInstance)
+				} else {
+					p.API.LogDebug("Failed to store JWT instance inside cloud OAuth instance data or there was no JWT instance installed previously.", "JWTInstance", instance.(*cloudOAuthInstance).JWTInstance)
```

Comment:
The "Failed to store JWT instance" piece of this seems alarming, especially since this is typically not an "error". Maybe something like this instead:

```suggestion
					p.API.LogDebug("No JWT instance inside cloud OAuth instance data.")
```